File: 백준/Bronze/2231. 분해합/분해합.py
n = int(input())

for i in range(1, n + 1):
    num = i
    result = 0
    while num > 0:
        result += num % 10
        num //= 10
    if result + i == n:
        print(i)
        break
    if i == n:
        print(0)

File: 백준/Bronze/2231. 분해합/분해합.cc
#include <iostream>
using namespace std;

int main() {
    int num;
    int result = 0;
    cin >> num;

    for(int i = 1; i < num; i++) {
        int sum = 0;
        int ans = i;
        while(ans != 0) {
            sum += ans % 10;
            ans /= 10;
        }
        if(sum + i == num) {
            result = i;
            break;
        }
    }

    cout << result << endl;

    return 0;
}